src/community/features-templating/features-templating-core/src/test/java/org/geoserver/featurestemplating/readers/JSONIncludesTest.java

```diff
@@ -174,6 +180,23 @@ public void testRecursionPingPong() {
         assertThat(ex.getMessage(), containsString("pong.json"));
     }
 
+    @Test
+    public void testIncludedModificationAreDetected() throws IOException, InterruptedException {
+        Resource resource = store.get("includeInObject.json");
+        RecursiveJSONParser parser = new RecursiveJSONParser(resource);
+        TemplateReaderConfiguration configuration =
+                new TemplateReaderConfiguration(new NamespaceSupport());
+        JSONTemplateReader reader =
+                new JSONTemplateReader(parser.parse(), configuration, parser.getWatchers());
+        RootBuilder rootBuilder = reader.getRootBuilder();
+        File file = resource.file();
```

Comment:
Seems like you're getting the resource of the primary file, rather than the one being included? 
Also, I guess a test for "merge" should be added too.

Reply:
fixed includeJson, added test for merge.